item-api/src/main/java/com/example/auction/item/api/UpdateItemResultCodes.java

```diff
@@ -0,0 +1,11 @@
+package com.example.auction.item.api;
+
+/**
+ *
+ */
+public enum UpdateItemResultCodes {
+    NOT_FOUND,
+    SUCCESS,
+    CAN_ONLY_UPDATE_DESCRIPTION,
+    CANT_UPDATE_AUCTION_IS_CLOSED;
```

Comment:
I don't think I'd use an enum here.  There's a few reasons:

1. There's only one success result. This is different to placing a bid, where there are multiple different success results, your bid may have been accepted (success) but was below the reserve, your bid may have been accepted (success) but was automatically outbid, and your bid may have been accepted (success) and is the current highest bid.
2. The UI is already going to render in such a way that it should only allow editing fields that are editable. So there's not a big need for describing exactly why it failed, the chance of a user encountering these errors is very low because the UI ensures this.
3. Most of these status can be expressed semantically in HTTP - NOT FOUND can be expressed by returning a 404 error, SUCCESS by returning 200, and the other two by returning some other 4xx error.  I think it's better to use semantic HTTP status codes than inventing a new protocol for returning the status.

By the way, you can easily return any status code with a message by throwing a TransportException, eg:

```java
throw new TransportException(TransportErrorCode.fromHttp(409), "Can only edit description once the auction has started")
```